src/rgw/rgw_asio_frontend.cc

```diff
@@ -214,9 +274,51 @@ int AsioFrontend::init()
   boost::system::error_code ec;
   auto& config = conf->get_config_map();
 
+#ifdef WITH_RADOSGW_BEAST_OPENSSL
+  // ssl configuration
+  auto cert = config.find("ssl_certificate");
+  const bool have_cert = cert != config.end();
+  if (have_cert) {
+    // only initialize the ssl context if it's going to be used
+    ssl_context = boost::in_place(ssl::context::tls);
```

Comment:
use of `ssl::context::tls` here disables the ssl v2 and v3 protocols. our s3tests currently pin the python requests library at 0.14.0, which can only do up to ssl v3. this causes some of the s3tests to fail against beast with `ssl handshake failed: wrong version number` errors. fixed in https://github.com/ceph/s3-tests/pull/217